### DISCUSSION TITLE
Why does the go redis/v8 version function input must have a context type?

### DISCUSSION THREAD

**Discussion by MrGlp** [redis/go-redis#1697] [Q&A] [answered]
Why does the go redis/v8 version function input must have a context type?
example: go redis/v8
func (c cmdable) Set(ctx context.Context, key string, value interface{}, expiration time.Duration) *StatusCmd {...}

but go redis does not
example: go redis
func (c cmdable) Set(key string, value interface{}, expiration time.Duration) *StatusCmd {...}

What is the difference and impact on use?

**Comment by vmihailenco** [comment, marked as answer]
context.Context is mainly used for canceling operations and tracing. It is a general Go feature so you should be able to google more info if needed.

**Comment by elgs** [comment]
That doesn't smell right.

**Reply by HeCorr** [reply]
if you don't care much about contexts, just throw a `context.TODO()` on there and call it a day.

**Reply by elgs** [reply]
> if you don't care much about contexts, just throw a `context.TODO()` on there and call it a day.

That's what I meant by ugly and smell. I switched redigo and called it a day.

**Reply by HeCorr** [reply]
> That's what I meant by ugly and smell. I switched redigo and called it a day.

that works too. I used to use redigo but switched to go-redis for the improved type-safety.

**Reply by nicksherron** [reply]
Maybe you should do some research and use your head instead of your nose. The context package was created for for the very reasons @vmihailenco mentioned. If you don't like the smell of context.Context you should probably not use go for anything http related, because its used extensively in net/http.

**Reply by elgs** [reply]
> Maybe you should do some research and use your head instead of your nose. The context package was created for for the very reasons @vmihailenco mentioned. If you don't like the smell of context.Context you should probably not use go for anything http related, because its used extensively in net/http.

Thumbs down. 👎

**Reply by nicksherron** [reply]
> > Maybe you should do some research and use your head instead of your nose. The context package was created for for the very reasons @vmihailenco mentioned. If you don't like the smell of context.Context you should probably not use go for anything http related, because its used extensively in net/http.
> 
> Thumbs down. 👎

Great rebuttal. I would respond but there is a ton of wise people who warn against arguing with stupid people and I am going to take their advice. 🏳
